projection/ibc_channel/view/ibc_denom_hash_mapping.go

```diff
@@ -0,0 +1,112 @@
+package view
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/crypto-com/chain-indexing/appinterface/rdb"
+)
+
+type IBCDenomHashMapping struct {
+	rdb *rdb.Handle
+}
+
+func NewIBCDenomHashMapping(handle *rdb.Handle) *IBCDenomHashMapping {
+	return &IBCDenomHashMapping{
+		handle,
+	}
+}
+
+func (ibcDenomHashMappingView *IBCDenomHashMapping) IfDenomExist(denom string) (bool, error) {
```

Comment:
Consider using `UPSERT`
https://stackoverflow.com/questions/4069718/postgres-insert-if-does-not-exist-already

Reply:
I think `UPSERT` is a more expensive operation compare to `EXISTS`. 

As we only need to insert the record for once, and the record is not supposed to be updated later.